server/createComponentRouter.js

```diff
@@ -0,0 +1,29 @@
+var React = require('react');
+var renderToString = require('react-dom/server').renderToString;
+var ReactRouter = require('react-router');
+
+var match = ReactRouter.match;
+var RoutingContext = ReactRouter.RoutingContext;
+
+module.exports = function createComponentRouter (routes) {
+  return function componentRouter (req, res, next) {
+    match({ routes: routes, location: req.url },
+        function (error, redirectLocation, renderProps) {
+
+      if (error) return res.status(500).send(error.message);
```

Comment:
> Also are there mixed tabs & spaces in this file? the indentation in the snippet above looks weird

It appears there's no mixed spaces and tabs.. but you're right. I'll fix the indentation.